object/object_test.go

```diff
@@ -1710,6 +1711,561 @@ func TestHandler_DeleteObjects(t *testing.T) {
+			},
+			args: args{
+				ctx: context.Background(),
+				request: &pb.MoveObjectRequest{
+					BucketSrc:  "notexistbucket",
+					BucketDest: "testbucket1",
+					KeySrc:     "file1",
+					KeyDest:    "newfile1",
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "source object doesnt exist",
+			fields: fields{
+				s3Client: s3Client,
+			},
+			args: args{
+				ctx: context.Background(),
+				request: &pb.MoveObjectRequest{
+					BucketSrc:  "testbucket",
+					BucketDest: "testbucket1",
+					KeySrc:     "notexistobject",
+					KeyDest:    "newfile1",
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "dest bucket is null",
+			fields: fields{
+				s3Client: s3Client,
+			},
+			args: args{
+				ctx: context.Background(),
+				request: &pb.MoveObjectRequest{
+					BucketSrc:  "testbucket",
+					BucketDest: "",
+					KeySrc:     "notexistobject",
+					KeyDest:    "newfile1",
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "source bucket is null",
+			fields: fields{
+				s3Client: s3Client,
+			},
+			args: args{
+				ctx: context.Background(),
+				request: &pb.MoveObjectRequest{
+					BucketSrc:  "",
+					BucketDest: "testbucket1",
+					KeySrc:     "notexistobject",
+					KeyDest:    "newfile1",
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "source key is null",
+			fields: fields{
+				s3Client: s3Client,
+			},
+			args: args{
+				ctx: context.Background(),
+				request: &pb.MoveObjectRequest{
+					BucketSrc:  "testbucket",
+					BucketDest: "testbucket1",
+					KeySrc:     "",
+					KeyDest:    "newfile1",
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "dest key is null",
+			fields: fields{
+				s3Client: s3Client,
+			},
+			args: args{
+				ctx: context.Background(),
+				request: &pb.MoveObjectRequest{
+					BucketSrc:  "testbucket",
+					BucketDest: "testbucket1",
+					KeySrc:     "notexistobject",
+					KeyDest:    "",
+				},
+			},
+			wantErr: true,
+		},
+	}
+
+	// Create connection to server
+	ctx := context.Background()
+	conn, err := grpc.DialContext(ctx, "bufnet", grpc.WithContextDialer(bufDialer), grpc.WithInsecure())
+	if err != nil {
+		t.Fatalf("Failed to dial bufnet: %v", err)
+	}
+	defer conn.Close()
+
+	// Create client
+	client := pb.NewUploadClient(conn)
+
+	// Iterate over test cases
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := client.MoveObject(tt.args.ctx, tt.args.request)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("Service.MoveObject() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+
+			// TODO: check if deepequal or cmp.equal
+			// reflect.DeepEqual is often incorrectly used to compare two like structs
+			// cmp.Equal is a better tool for comparing structs.
+			// https://golang.org/pkg/reflect/#DeepEqual
+			if got != nil && !cmp.Equal(tt.want, got, cmpopts.IgnoreUnexported(pb.MoveObjectResponse{})) {
+				t.Errorf("UploadHandler.MoveObject() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
```

Comment:
What about a test for dest bucket doesn't exist?
And key already exists in dest bucket?

Reply:
f the destination bucket does not exist, a new bucket is created.

Reply:
The building of the object's key is in the file-service. The create/move function relies on the key sent by the file-service.